src/index.js

```diff
@@ -16,7 +28,24 @@ const config = {
 };
 firebase.initializeApp(config);
 
+
+// This is unique, it's probably one of the few actions we'll fire outside the
+// context of the webapp, but only because firebase's auth works in this semi-
+// global way.
+firebase.auth().onAuthStateChanged(function(user) {
+  // This is going to be weird initially because firebase handles this for us
+  // but it will get across how redux wants you to think about actions etc
+  if(user) {
+    store.dispatch(handleUserAuthed(user));
+  } else {
+    store.dispatch(notAuthed());
+  }
+})
+
+
 ReactDOM.render(
-  <App />,
+  <Provider store={store}>
```

Comment:
`<Provider>` is a magical component that makes the redux store available to all children of `<App>` using a special method `connect()` that comes with `react-redux`. we'll see it deployed in a subsequent pr anyway.